### DISCUSSION TITLE
Empty message.content using a user bot

### DISCUSSION THREAD

**Discussion by zouhaofan** [Rapptz/discord.py#6850] [Q&A] [answered]
Hi guys, I'm getting empty content of messages which aren't sent by myself. But my bot can still get the content of all messages. Are you having this problem too? How should I solve it?

**Comment by esmaycat** [comment, marked as answer]
> On April 30th, 2021 discord made some change that broke receiving message content and embeds (and maybe more) on selfbots only. If you're receiving this issue, then you're using a selfbot, which is against the discord TOS... it is also deprecated by d.py since 1.7 and won't receive support. You need to change to a real bot if you want support

In answer to your question it is not solvable.

There was also an issue which described this problem at #6820.
